internal/alertmanager/currentalert/process.go

```diff
@@ -238,8 +238,56 @@ func addPvc(log logr.Logger, alertLabels model.LabelSet, alertAnnotations model.
 	return nil
 }
 
-func resizePvc(labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
-	//TODO
+func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
+	cr, err := k8sutil.GetCr(string(labels["kafka_cr"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	pvc, err := getPvc(string(labels["persistentvolumeclaim"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	for i, broker := range cr.Spec.Brokers {
+		if strconv.Itoa(int(broker.Id)) == pvc.Labels["brokerId"] {
+			var storageConfigs []v1beta1.StorageConfig
+
+			if broker.BrokerConfig == nil {
+				storageConfigs = cr.Spec.BrokerConfigGroups[cr.Spec.Brokers[i].BrokerConfigGroup].StorageConfigs
+			} else {
+				storageConfigs = broker.BrokerConfig.StorageConfigs
+			}
+
+			for i, c := range storageConfigs {
+				if c.MountPath == pvc.Annotations["mountPath"] {
+					size := *c.PvcSpec.Resources.Requests.Storage()
+
+					size.Add(resource.MustParse(string(annotiations["diskSize"])))
+
+					c.PvcSpec.Resources.Requests = corev1.ResourceList{
+						"storage": size,
+					}
+
+					if broker.BrokerConfig == nil {
+						broker.BrokerConfig = &v1beta1.BrokerConfig{
+							StorageConfigs: []v1beta1.StorageConfig{c},
+						}
+					} else {
+						broker.BrokerConfig.StorageConfigs[i] = c
```

Comment:
Wouldn't this wipe out the fields of existing storage config other than `storage` field (e.g. `mountPath`)? `mountPath` doesn't have `omitempty` specified in the json tag so it may wipe out `mountPath` if no value is set.

Instead of  `broker.BrokerConfig.StorageConfigs[i] = c` it's safer to update the `storage` field of the existing storage config instance if exists.

Reply:
I played around with this a lot, and I think I got it simpler and less possibly dangerous but to explain why it looks the way it does:

First I have to resolve what is the `brokerConfig`/`[]storageConfig` for the the broker in question which is what I do here https://github.com/banzaicloud/kafka-operator/blob/101bc86d972cd59983cce899a8bcee81b3e73f17/internal/alertmanager/currentalert/process.go#L256

namely:
```
			if broker.BrokerConfig == nil {
				storageConfigs = cr.Spec.BrokerConfigGroups[cr.Spec.Brokers[i].BrokerConfigGroup].StorageConfigs
			} else {
				storageConfigs = broker.BrokerConfig.StorageConfigs
			}
```

Now that I have some storageConfig slice I go through them looking for a `mountPath` that matches (https://github.com/banzaicloud/kafka-operator/blob/101bc86d972cd59983cce899a8bcee81b3e73f17/internal/alertmanager/currentalert/process.go#L263) if I find it I resize the PVC based on the `incrementBy` set in the alert. 

Where it gets tricky is in the original CR request the brokerConfig could still be null or not so I try to handle the case here where I create a `brokerConfig` with just a `storageConfig` in it or I update the specific `storageConfig` if one existed. I update it from the copy the `range` made of object in the slice + my modification to the pvc size (https://github.com/banzaicloud/kafka-operator/blob/101bc86d972cd59983cce899a8bcee81b3e73f17/internal/alertmanager/currentalert/process.go#L277)

```
if broker.BrokerConfig == nil {
  broker.BrokerConfig = &v1beta1.BrokerConfig{
    StorageConfigs: []v1beta1.StorageConfig{c},
  }
} else {
  broker.BrokerConfig.StorageConfigs[i] = c
}
```

since its a copy of the object was in the slice previously it should contain all that was there previously such as `mountPath` or anything else (I think the storage config only has `mountPath` and `PvcSpec` 

then in https://github.com/banzaicloud/kafka-operator/blob/101bc86d972cd59983cce899a8bcee81b3e73f17/internal/alertmanager/currentalert/process.go#L286 I update the original CR I got, I wanted to just update the storageConfig but since a brokerConfig might not exist I have to update the whole brokerConfig object. So because the config might not exist I can't safely do something like `cr.Spec.Brokers[i].BrokerConfig.StorageConfig[ii].PvcSpec.Resources.Requests = newPvcConfig{}` as some or none of that whole path might exist. I could `if/else` it until I find the lowest level of the path and update just that, but I think my current solution balances safety and readability. 

I do cover some of the cases in the unit test I wrote, making sure we don't loose a storage config during the update. 

If you have other suggestions I'm happy to try.